pkg/api/unversioned/deep_copy_generated.go

```diff
@@ -21,127 +21,15 @@ limitations under the License.
 package unversioned
 
 import (
-	"time"
-
 	conversion "k8s.io/kubernetes/pkg/conversion"
+	time "time"
 )
 
-func DeepCopy_unversioned_APIGroup(in APIGroup, out *APIGroup, c *conversion.Cloner) error {
```

Comment:
We definitely want to make it explicit.  I'm ok with polluting a wild
carded list of packages with the generated packages (k8s.io/kubernetes/*) -
I think that's easier to reason about when we start layering other
generation scopes outside of Kube.

On Mon, May 23, 2016 at 11:52 AM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/api/unversioned/deep_copy_generated.go
> <https://github.com/kubernetes/kubernetes/pull/26051#discussion_r64243536>
> :
>
> >  )
> >
> > -func DeepCopy_unversioned_APIGroup(in APIGroup, out *APIGroup, c *conversion.Cloner) error {
>
> In playing with making this not be checked in, I started using a tag (//
> +k8s:gen_deep_copy=true) in packages that want deep copies. We could do
> that, but change the value from true to "true,register" indicating whether
> or not to register with scheme.
>
> This is more explicit and obvious, and doesn't require knowing where the
> edges are, but it does require that deep-copy be explicitly generated for
> "helper" packages which is somewhat polluting for them (but not worse than
> dropping a file into them).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26051/files/ee463c073a7f6853fc2a6019de435aa5d4e6c2c4#r64243536>
>